k8s.gcr.io/Vanity-Domain-Flip.md

```diff
@@ -0,0 +1,191 @@
+has been agreed that the community should use it as the new place to push
+production images (instead of `gcr.io/google-containers`). **We can solve the
+above problem by flipping the vanity domain (`k8s.gcr.io`) from
+`gcr.io/google-containers` to `gcr.io/k8s-artifacts-prod`**. This way, no change
+needs to be made in the Kubernetes codebase.
+
+The minimum prerequisite is that the existing images in `google-containers` must
+be copied into `k8s-artifacts-prod` in order to ensure that the domain flip
+happens transparently without incurring any interruptions. However there are
+other infrastructural improvements that the community has designed, such as
+explicit backups, disaster recovery, and also auditing and alerting.
+
+The rest of this document explains the infrastructural improvements surrounding
+`gcr.io/k8s-artifacts-prod`.
+
+# How New Images Get Promoted (Pushed) to Production
+
+To get new images into `gcr.io/google-containers`, a Googler must approve a
+change in Google's private repository.
+
+On the other hand, `gcr.io/k8s-artifacts-prod` is integrated with a
+publicly-visible GitHub repository, named [k8s.io][k8sio]. The [promoter][CIP]
+watches this repository for changes and promotes images. In addition, a system
+of setting up staging repos, and promoting from them into
+`gcr.io/k8s-artifacts-prod` has been [created][staging-subproject] so that owners of
+subprojects in the community can take control of how their images are released.
+
+## The Promoter (cip)
+
+The [Container Image Promoter][CIP] (henceforth "the promoter") is the OSS
+rewrite of the [promoter used internally within Google][internal-promoter]. It
+works by reading in a set of promoter manifests (YAMLs) that describe the
+desired state of a Docker registry's image contents, and proceeds to copy in any
+missing images. Currently the toplevel `k8s.gcr.io` directory at the [k8s.io
+Github repo][k8sio] defines such a set of promoter manifests.
+
+## Prow Integration
+
+The act of invoking the promoter as a postsubmit against the k8s.io repo is done
+by [Prow][prow], as the `post-k8sio-cip` Prow job. There are other Prow jobs
+that integrate with the promoter, and the ones relevant to this doc are outlined
+in the table below:
+
+- [`post-k8sio-cip`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml)
+  Postsubmit job against k8s.io repo holding promoter manifests. The promoter
+  manifests here are those that promote from the various staging subproject
+  repos to `gcr.io/k8s-artifacts-prod/<subproject>/<image>`. It uses the
+  `k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com` service
+  account to write to `{asia,eu,us}.gcr.io/k8s-artifacts-prod`. For all
+  intents and purposes, **this is the gatekeeper for new images going into
+  `k8s-artifacts-prod`**.
+- [`ci-k8sio-cip`](https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml)
```

Comment:
ditto on logs

Reply:
Why is this called "ci-..." rather than something like "periodic-...."

Reply:
> Why is this called "ci-..." rather than something like "periodic-...."

IIRC this convention of using the "ci-" prefix is just how most jobs in test-infra are named. See https://github.com/kubernetes/test-infra/pull/11414#discussion_r270205042